bin/pyang

```diff
@@ -252,12 +259,22 @@ Validates the YANG module in <filename> (or stdin), and all its dependencies."""
     ctx.max_status = o.max_status
 
     # make a map of features to support, per module
+    if o.features and o.exclude_features:
+        sys.stderr.write("can't specify both --features and "
+                         "--exclude-features\n")
+        sys.exit(1)
     if o.hello:
         for mn, rev in hel.yang_modules():
             ctx.features[mn] = hel.get_features(mn)
     for f in ctx.opts.features:
         (modulename, features) = parse_features_string(f)
         ctx.features[modulename] = features
+    for f in ctx.opts.exclude_features:
+        (modulename, features) = parse_features_string(f)
+        # unlike --features, --exclude-features only does anything if
+        # individual features are listed explicitly
+        if features:
+            ctx.features[modulename] = ['!' + f for f in features]
```

Comment:
Now when I see the implementation, I think that perhaps it would be better to set `ctx.excluded_features` in this case.
Then the code in statements.py would be simpler.
You should still keep the code that prepares `ctx.features` for plugins - in this case both `ctx.features` and `ctx.exclude_features` would be set, for plugins (but not for the main statements.py validation).

Reply:
Hmm, I take your point, but that wouldn't be backwards-compatible from the point of view of the plugin. Currently a plugin can assume that `ctx.features`(if defined for a given module) is the enabled features, but this would no longer be the case.

BTW, every time I see `['!' + f for f in features]` (for which I am to blame!) I want to change it to `[('!' + f) for f in features]` or `['!%s' % f for f in features]`.

Reply:
I meant that you should keep the code that sets `ctx.features` for plugins.  They would also see (if the look) `ctx.exclude_features`, but that's ok.

But note that a plugin might install a validation fun for example, and when that fun is invoked it has to be prepared for this new meaning (either be prepared for `!foo` or the `ctx.exclude_features`.

Reply:
> But note that a plugin might install a validation fun for example

I hadn't considered this. I agree that we have to keep `ctx.features` unchanged, and add a new `ctx.exclude_features`.

An old plugin wouldn't know to look at `ctx.exclude_features` but none of the standard plugins look at `ctx.features` so we're only talking (theoretically) about third-party plugins. So I agree that we can live with this!